include/arch/arm/aarch32/thread.h

```diff
@@ -74,25 +74,46 @@ struct _thread_arch {
 	struct _preempt_float  preempt_float;
 #endif
 
-#if defined(CONFIG_USERSPACE) || defined(CONFIG_FPU_SHARING)
+#if defined(CONFIG_ARM_STORE_EXC_RETURN)
 	/*
 	 * Status variable holding several thread status flags
 	 * as follows:
 	 *
-	 * +--------------bit-3----------bit-2--------bit-1---+----bit-0------+
+	 * byte 0
+	 * +-bits 4-7-----bit-3----------bit-2--------bit-1---+----bit-0------+
 	 * :          |             |              |          |               |
-	 * : reserved |<Guard FLOAT>| <FP context> | reserved |  <priv mode>  |
-	 * :   bits   |             | CONTROL.FPCA |          | CONTROL.nPRIV |
+	 * : reserved |<Guard FLOAT>|   reserved   | reserved |  <priv mode>  |
+	 * :   bits   |             |              |          | CONTROL.nPRIV |
+	 * +------------------------------------------------------------------+
+	 *
+	 * byte 1
+	 * +----------------------------bits 8-15-----------------------------+
+	 * :              Least significant byte of EXC_RETURN                |
+	 * : bit 15| bit 14| bit 13 | bit 12| bit 11 | bit 10 | bit 9 | bit 8 |
+	 * :  Res  |   S   |  DCRS  | FType |  Mode  | SPSel  |  Res  |  ES   |
 	 * +------------------------------------------------------------------+
 	 *
 	 * Bit 0: thread's current privileged mode (Supervisor or User mode)
 	 *        Mirrors CONTROL.nPRIV flag.
-	 * Bit 2: indicating whether the thread has an active FP context.
+	 * Bit 2: Deprecated in favor of FType. Note: FType = !CONTROL.FPCA.
+	 *        indicating whether the thread has an active FP context.
 	 *        Mirrors CONTROL.FPCA flag.
 	 * Bit 3: indicating whether the thread is applying the long (FLOAT)
 	 *        or the default MPU stack guard size.
+	 *
+	 * Bits 8-15: Least significant part of the EXC_RETURN value, which is
+	 *            found in LR upon entering the pendsv handler.
 	 */
-	uint32_t mode;
+	union {
+		uint32_t mode;
+
+		__packed struct {
+			uint8_t bits;
```

Comment:
I would keep the "mode" in the name - this is ultimately, a mode status variable.
Would use
```
mode_flags
mode_exc_return
mode_reserved
```